### DISCUSSION TITLE
Can I suspend plugin activity if app is not in focus?/How are 'file-edits' detected

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [Vinzent03/obsidian-git#746] [Problems]
Hello,

I have a bit of a peculiar issue right now.
I have a vault I sync via obsidian-git, but it also hosts a [quarto-website-project](https://quarto.org/docs/websites/). 

This means I use both VS Code and Obsidian for working - Obsidian for writing, VS Code for scripting, automation, ...
While I work in VSCode, I often make specific commits which change project-structure, update configuration or script-files. These commits are hand-curated.

At the same time, Obsidian is running in the background, with the `obsidian-git`-plugin, and this poses some problems. I have the plugin configured to run 5 minutes after last change occured.

This poses a unique issue, because while selectively managing the project in VSCode, `obsidian-git` can end up committing and pushing stuff if I take too long without saving a file in VSC manually. 

Would it be possible to only trigger the execution-routine if the obsidian-application is active/is in focus? In principle such an option would solve this issue  - but I am also completely aware that this is a super niche issue.

Thank you.
Sincerely,
~Gw

**Comment by Vinzent03** [comment]
I definitely want the backup to be run after the 5 minutes, even if Obsidian is in background. I could only think of ignoring modify events if Obsidian is in background, but I'm unsure whether this is a good idea or not. And probably wouldn't even solve your issue, right? So I don't think there's a solution to your problem other than closing Obsidian while doing that work or increasing the timers.

**Reply by Gewerd-Strauss** [reply]
Hmm, alright. 

I was expecting this answer cuz I couldn't think of anything else myself but wanted to make sure. Thank you. I suppose I will just have to deal with it then.

**Reply by Gewerd-Strauss** [reply]
Wait, a small thought: 

Can one change the 'set to 0 to disable' to a checkbox? And then register a hotkey to toggle this manually? 

That would be _miles_ faster than the current solution of `go to settings>community-plugins`>`search for git`>`disable plugin`>`do work in VSC`>`pray to remember to re-enable the plugin>`.


If a hotkey'able solution would be implemented I don't think the 'set to 0 to disable' would be sustainable, thus the suggestion to move to a checkbox.

**Reply by Gewerd-Strauss** [reply]
I have since created a rudimentary solution which suffices for me, can be found under  [Gewerd-Strauss/obsidian-git@feat/suggested-solution-to-#746](https://github.com/Gewerd-Strauss/obsidian-git/tree/feat/suggested-solution-to-%23746). I don't think this is the best way to do this; but on my end my bit of testing today couldn't break it and it works on my end.

Note that I have deliberately not changed version numbers or other metadata as I am unaware if any conventions would exist - and since currently this is only a private modification, I don't immediately see the need to do so.

**Reply by Gewerd-Strauss** [reply]
Sorry to revive this and ping you @Vinzent03, but could I ask your opinion on the solution I outlined above? 

I've since tested this extensively, and so far this has worked flawlessly on my end. For simplicity's sake, I suppose 

https://github.com/Vinzent03/obsidian-git/compare/master...Gewerd-Strauss:obsidian-git:feat/suggested-solution-to-%23746

would be the easiest way to get an overview on what I propse.

I am currently hesitant to write a PR solely because I am running the modifications on version 2.27.0 of the plugin; and don't have the time to re-integrate them into 2.30.1 right now (as I am in the process of writing a thesis right now). However, at a short glance I don't see any immediate red flags myself.

And thus I would like to know whether or not you would even be interested in this feature/its merge into the official repo to begin with.  
(That is, the ability to suppress commit-creation locally via setting/hotkey, but without requiring to disable the plugin altogether. This works wonder if you work on a project with automated commits made by the `obsidian-git`-plugin, where you still require regular control over specific changes and have the need to commit them manually in a curated manner)


Thank you.
Sincerely,
~Gw

**Comment by strazto** [comment]
I have a similar use-case - I sometimes edit things from, eg, my IDE (dataviewjs for example), and I don't want the same commit & sync interval in that case.
What would give me the most confidence is for app focus events (stop actively editing, focus a different window for a certain time, etc) to trigger syncs, and then a reasonable time-based approach if the app is in focus
